src/main/java/flaxbeard/steamcraft/client/render/ItemSteamToolRenderer.java

```diff
@@ -35,65 +40,61 @@ public boolean shouldUseRenderHelper(ItemRenderType type, ItemStack item, ItemRe
+        for (int pass = 0; pass < item.getItem().getRenderPasses(item.getItemDamage()); pass++) {
+            GL11.glEnable(GL11.GL_ALPHA_TEST);
+            IIcon icon = item.getItem().getIcon(item, pass);
+            int color = item.getItem().getColorFromItemStack(item, pass);
+            // 16,777,215 is the "white" default returned when the pass does not render a drill head.
+            if (color != 16777215) {
+                float r = (float)(color >> 16 & 255) / 255.0F;
+                float g = (float)(color >> 8 & 255) / 255.0F;
+                float b = (float)(color & 255) / 255.0F;
+                GL11.glColor4f(r, g, b, 1.0F);
+            }
+            renderItem.renderIcon(0, 0, icon, 16, 16);
+            GL11.glDisable(GL11.GL_ALPHA_TEST);
         }
 
-        GL11.glEnable(GL11.GL_ALPHA_TEST);
-        renderItem.renderIcon(0, 0, icon, 16, 16);
-        GL11.glDisable(GL11.GL_ALPHA_TEST);
-
-        if (player != null && player instanceof EntityPlayer) {
-            ExtendedPropertiesPlayer nbt = (ExtendedPropertiesPlayer)
-              player.getExtendedProperties(Steamcraft.PLAYER_PROPERTY_ID);
-            int use = 0;
-            switch (toolType) {
-                case 0: {
-                    use = nbt.drillInfo.right;
-                    break;
-                }
-                case 1: {
-                    use = nbt.axeInfo.right;
-                    break;
-                }
-                case 2: {
-                    use = nbt.shovelInfo.right;
-                    break;
-                }
+        int use = 0;
+        switch (this.type) {
+            case DRILL: {
+                use = ItemSteamDrill.checkNBT(item).getInteger("Speed");
+                break;
             }
-
-            double health = (1000.0D - use) / 1000.0D;
-            if (use > 0 && itemStack == Minecraft.getMinecraft().thePlayer.getHeldItem()) {
-                GL11.glPushMatrix();
-                int j1 = (int) Math.round(13.0D - health * 13.0D);
-                int k = (int) Math.round(255.0D - health * 255.0D);
-                GL11.glDisable(GL11.GL_LIGHTING);
-                GL11.glDisable(GL11.GL_DEPTH_TEST);
-                GL11.glDisable(GL11.GL_TEXTURE_2D);
-                GL11.glDisable(GL11.GL_ALPHA_TEST);
-                GL11.glDisable(GL11.GL_BLEND);
-                Tessellator tessellator = Tessellator.instance;
-                int l = 255 - k << 16 | k << 8;
-                int i1 = (255 - k) / 4 << 16 | 16128;
-                this.renderQuad(tessellator, 2, 10, 13, 2, 0);
-                this.renderQuad(tessellator, 2, 10, 12, 1, i1);
-                this.renderQuad(tessellator, 2, 10, j1, 1, l);
-                GL11.glEnable(GL11.GL_TEXTURE_2D);
-                GL11.glEnable(GL11.GL_LIGHTING);
-                GL11.glEnable(GL11.GL_DEPTH_TEST);
-                GL11.glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
-                GL11.glPopMatrix();
+            case AXE: {
```

Comment:
I could probably make checkNBT a function that is not dependent on the item type (in UtilSteamTool or SteamToolHelper), so this switch statement would not exist anymore (cc @Lance5057 )